src/utils.rs

```diff
@@ -41,3 +41,36 @@ pub fn handle_communication_err(peer_addr: SocketAddr, e: &Error, details: &str)
     );
     let _ = ctx_mut(|c| c.connections.remove(&peer_addr));
 }
+
+#[cfg(test)]
+pub mod testing {
+    use crate::dirs::{Dirs, OverRide};
+    use crate::NodeInfo;
+    use rand::Rng;
+    use std::env;
+    use std::net::{Ipv4Addr, SocketAddr, SocketAddrV4};
+    use std::path::PathBuf;
+
+    pub fn test_dirs() -> Dirs {
+        Dirs::Overide(OverRide::new(&unwrap!(tmp_dir().to_str())))
+    }
+
+    pub fn rand_peer() -> NodeInfo {
+        let peer_cert_der =
+            rcgen::generate_simple_self_signed(vec!["Test".to_string()]).serialize_der();
```

Comment:
Can't we use `Config::with_default_cert()` and not use rc-gens and the rest as that does all that internally, unless the intention was to create a special Certificate - which doens't seem so here as it's just a random one.

Reply:
How about `SerialisableCertificate::default()` since I don't need anything else from the config.